Add check for enabled countries in cart and order

PrestaEdit opened the pull request:
Introduces Cart::checkCountriesAreEnabled() to ensure both delivery and invoice countries are active. Integrates this check into CartController and OrderController to prevent checkout with disabled countries, improving validation and user feedback.

<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Check that countries are enabled before using them in a order validation.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| UI tests     |https://github.com/SiraDIOP/ga.tests.ui.pr/actions/runs/18874820050
| How to test?      | See #39786 and check that you will not be able to reach payment step.
| Fixed issue or discussion?     | Fixes #39786

PrestaEdit committed:
Add check for enabled countries in cart and order

Introduces Cart::checkCountriesAreEnabled() to ensure both delivery and invoice countries are active. Integrates this check into CartController and OrderController to prevent checkout with disabled countries, improving validation and user feedback.

Hlavtox commented on controllers/front/OrderController.php:
```diff
@@ -284,7 +285,8 @@ public function initContent(): void
             || $this->context->cart->checkAllProductsAreStillAvailableInThisState() !== true
-            || $this->context->cart->checkAllProductsHaveMinimalQuantities() !== true) {
+            || $this->context->cart->checkAllProductsHaveMinimalQuantities() !== true
+            || $this->context->cart->checkCountriesAreEnabled() !== true) {
```
I would put it outside of this `if`, for clarity.

PrestaEdit committed:
Refactor order validation checks in OrderController

Separated the check for enabled countries from other cart validation checks in the order process. This improves clarity and ensures address validation is handled distinctly from product-related checks.

SiraDIOP reviewed (commented):
Hello @PrestaEdit, 

Thanks for your PR, i tested it and it's good for me :), i'm waiting automated tests to approve this PR

Uploading Enregistrement de l’écran 2025-10-24 à 10.23.21.mov…
<img width="1433" height="756" alt="Capture d’écran 2025-10-28 à 17 00 24" src="https://github.com/user-attachments/assets/d846334d-b80a-47c7-97ee-706095b9d0e4" />


Thanks

nicosomb commented:
all is green, thank you @PrestaEdit and @SiraDIOP !
